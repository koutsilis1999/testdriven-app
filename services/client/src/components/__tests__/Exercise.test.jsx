import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";

import AceEditor from "react-ace";
jest.mock("react-ace");

import Exercise from "../Exercise";

const testData = {
  exercise: {
    id: 0,
    body: `Define a function called sum that takes two integers
      as arguments and returns their sum.`,
  },
  editor: {
    value: "# Enter your code here.",
    button: {
      isDisabled: false,
    },
    showGrading: false,
    showCorrect: false,
    showIncorrect: false,
  },
  isAuthenticated: false,
  onChange: jest.fn(),
  submitExercise: jest.fn(),
};

test("Exercise renders properly", () => {
  const wrapper = shallow(<Exercise {...testData} />);
  const heading = wrapper.find("h5");
  expect(heading.length).toBe(1);
  expect(heading.text()).toBe(testData.exercise.body);
});

test("Exercise renders a snapshot properly when not authenticated", () => {
  const tree = renderer.create(<Exercise {...testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Exercise renders a snapshot properl when authenticated", () => {
  testData.isAuthenticated = true;
  const tree = renderer.create(<Exercise {...testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
