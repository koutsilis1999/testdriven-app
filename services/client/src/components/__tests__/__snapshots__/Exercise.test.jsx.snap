// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise renders a snapshot properl when authenticated 1`] = `
<div>
  <h5
    className="title is-5"
  >
    Define a function called sum that takes two integers
      as arguments and returns their sum.
  </h5>
  <div>
    <button
      className="button is-primary"
      disabled={false}
      onClick={[Function]}
    >
      Run Code
    </button>
  </div>
  <br />
  <br />
</div>
`;

exports[`Exercise renders a snapshot properly when not authenticated 1`] = `
<div>
  <h5
    className="title is-5"
  >
    Define a function called sum that takes two integers
      as arguments and returns their sum.
  </h5>
  <br />
  <br />
</div>
`;
